packages/api/src/menu/submenu.ts

```diff
@@ -46,8 +48,14 @@ export function itemFromKind([rid, id, kind]: [number, string, ItemKind]):
   /* eslint-enable @typescript-eslint/no-unsafe-return */
 }
 
-export type SubmenuOptions = Omit<MenuItemOptions, 'accelerator' | 'action'>
-  & MenuOptions
+export type SubmenuOptions = (Omit<MenuItemOptions, 'accelerator' | 'action'>
+  & MenuOptions) & {
+  /**
+   * Icon to be used for the submenu.
+   * Note: you may need the `image-ico` or `image-png` Cargo features to use this API.
+   */
+  icon?: NativeIcon | string | Image | Uint8Array | ArrayBuffer | number[]
```

Comment:
Maybe we can extract this to a type alias now (something like `MenuIcon`) with this being used in multiple places

Reply:
I moved the type to packages/api/src/image.ts. If it needs to be relocated elsewhere, please let me know.